src/cowrie/commands/wget.py

```diff
@@ -189,41 +185,55 @@ def success(self, data, outfile):
                                   shasum=self.artifactFile.shasum)
 
         # Update honeyfs to point to downloaded file or write to screen
-        if outfile != '-':
-            self.fs.update_realfile(self.fs.getfile(outfile), self.artifactFile.shasumFilename)
-            self.fs.chown(outfile, self.protocol.user.uid, self.protocol.user.gid)
+        if self.outfile != '-':
+            self.fs.update_realfile(self.fs.getfile(self.outfile), self.artifactFile.shasumFilename)
+            self.fs.chown(self.outfile, self.protocol.user.uid, self.protocol.user.gid)
         else:
             with open(self.artifactFile.shasumFilename, 'rb') as f:
                 self.writeBytes(f.read())
 
         self.exit()
 
     def error(self, error, url):
-        if hasattr(error, 'getErrorMessage'):  # exceptions
-            errorMessage = error.getErrorMessage()
-            self.errorWrite(errorMessage + '\n')
-            # Real wget also adds this:
-        if hasattr(error, 'webStatus') and error.webStatus and hasattr(error, 'webMessage'):  # exceptions
-            self.errorWrite('{} ERROR {}: {}\n'.format(time.strftime('%Y-%m-%d %T'), error.webStatus.decode(),
-                                                       error.webMessage.decode('utf8')))
+        # we need to handle 301 redirects separately
+        if hasattr(error, 'webStatus') and error.webStatus.decode() == '301':
+            self.errorWrite('{} {}\n'.format(error.webStatus.decode(), error.webMessage.decode()))
+            https_url = error.getErrorMessage().replace('301 Moved Permanently to ', '')
+            self.errorWrite('Location {} [following]\n'.format(https_url))
+
+            # do the download again with the https URL
+            self.deferred = self.download(https_url.encode('utf8'), self.outfile)
+            if self.deferred:
+                self.deferred.addCallback(self.success)
+                self.deferred.addErrback(self.error, https_url)
+            else:
+                self.exit()
```

Comment:
If the web server returns a HTTP 301 redirect, we end up in error. In this case we need to stop the error propagation and call download again using the https URL returned with redirect.